bindings/rust/src/init.rs

```diff
@@ -0,0 +1,139 @@
+    let mut implementations = HashMap::new();
+
+    for target_dir in embed::CONFIGS_DIR.dirs() {
+        let target_name = target_dir.path().file_name().unwrap().to_str().unwrap();
+        let target_data = target_dir
+            .get_file(format!("{}/superchain.yaml", target_name))
+            .expect("Failed to find superchain.yaml config file")
+            .contents();
+        let entry_config: SuperchainConfig = serde_yaml::from_slice(target_data)
+            .expect("Failed to deserialize superchain.yaml config file");
+
+        let mut chain_ids = Vec::new();
+        for chain in target_dir.entries() {
+            let chain_name = chain.path().file_name().unwrap().to_str().unwrap();
+            if !is_config_file(chain_name) {
+                continue;
+            }
+
+            let chain_data = chain.as_file().unwrap().contents();
+            let mut chain_config: ChainConfig = serde_yaml::from_slice(chain_data)
+                .expect("Failed to deserialize chain config file");
+            chain_config.chain = chain_name.trim_end_matches(".yaml").to_string();
+
+            chain_config.set_missing_fork_configs(&entry_config.hardfork_defaults);
+
+            let json_file_name = chain_config.chain.clone() + ".json";
+            let addresses_data = embed::EXTRA_DIR
+                .get_file(format!("addresses/{}/{}", target_name, json_file_name))
+                .expect("Failed to find address list file")
+                .contents();
+            let addrs: AddressList = serde_json::from_slice(addresses_data)
+                .expect("Failed to deserialize address list file");
+
+            let genesis_config_data = embed::EXTRA_DIR
+                .get_file(format!(
+                    "genesis-system-configs/{}/{}",
+                    target_name, json_file_name
+                ))
+                .expect("Failed to find genesis system config file")
+                .contents();
+            let genesis_config: GenesisSystemConfig = serde_json::from_slice(genesis_config_data)
+                .expect("Failed to deserialize genesis system config file");
+
+            let id = chain_config.chain_id;
+            chain_config.superchain = target_name.to_string();
+            genesis_system_configs.insert(id, genesis_config);
+            op_chains.insert(id, chain_config);
+            addresses.insert(id, addrs);
+            chain_ids.push(id);
+        }
+
+        // match target_name {
+        //     "mainnet" => {
+        //         if let Ok(ci_mainnet_rpc) = std::env::var("CIRCLE_CI_MAINNET_RPC") {
+        //             entry_config.l1.public_rpc = ci_mainnet_rpc;
+        //         }
+        //     }
+        //     "sepolia" | "sepolia-dev-0" => {
+        //         if let Ok(ci_sepolia_rpc) = std::env::var("CIRCLE_CI_SEPOLIA_RPC") {
+        //             entry_config.l1.public_rpc = ci_sepolia_rpc;
+        //         }
+        //     }
+        //     _ => {}
+        // }
```

Comment:
@refcell any tips on how to read env vars when running in no_std? this is the only thing left out because of it.

Reply:
Can we just introduce an `std` feature flag here and run ci tests with std enabled?

Reply:
yes good idea!